src/System.Linq/tests/SingleTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Linq.Tests.Helpers;
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public class SingleTests
+    {
+        [Fact]
+        public void FailOnEmpty()
+        {
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<int>().Single());
+        }
+
+        [Fact]
+        public void DefaultOnEmpty()
+        {
+            Assert.Equal(0, Enumerable.Empty<int>().SingleOrDefault());
+        }
+
+        [Fact(Skip="Runs for thousands of years. Then throws wrong exception")]
+        public void FailOnManyMatches()
+        {
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Range(0, int.MaxValue).Single(i => i > 0));
```

Comment:
Thanks, @hackcraft.

> I'm more used to workflows where one doesn't test for behaviour one hasn't decided is or isn't desirable

Yup, completely understand.  We have to worry about an additional category, though, that of backwards compatibility.  That's lessened at the moment for .NET Core, but it's still an issue, and we'll often have tests that verify the current behavior regardless of whether it's a desirable behavior, because someone could have taken a dependency on it and we at least want to know and be thoughtful about any changes that could have a visible behavioral impact.